sql/tests: TestRandomSyntaxSQLSmith failed

sql/tests.TestRandomSyntaxSQLSmith [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/20377459?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/20377459?buildTab=artifacts#/) on master @ [82759997da526f08ead6de93e47fb30819edf10e](https://github.com/cockroachdb/cockroach/commits/82759997da526f08ead6de93e47fb30819edf10e):

Random syntax error:

```
    rsg_test.go:982: Crash detected: server panic: pq: internal error: expected *DLTree, found tree.dNull
```
Query:

```
        SELECT
        	tab_17578._int4 AS "'col_43495",
        	"t̄a%pb_17579".crdb_internal_origin_id AS c̱ol_43496,
        	'2011-08-04 10:22:11.000027':::TIMESTAMP AS 😎col_43497,
        	"t̄a%pb_17579"._enum AS " c\\u7487ol_43498",
        	"t̄a%pb_17579"._uuid AS col_43499,
        	"t̄a%pb_17579"._string AS col_43500,
        	"t̄a%pb_17579"._bool AS col_43501,
        	NULL AS col_43502,
        	'\x':::BYTES AS col_43503,
        	(
        		WITH
        			with_7149 (col😸_43504)
        				AS (
        					SELECT
        						*
        					FROM
        						(
        							VALUES
        								('QoHL.zG'),
        								(
        									((NULL::LTREE || (ARRAY['','VgmDw.YYkxeQY.GOQW.gFzzkVcFp.xCxxs.xyXgy.SfOtEmQag.dMQEXqo','xh','TN.c.yuNdfPNx.if.sLDrHU.qYO.OrNT','']::LTREE[] || ARRAY['IMnXAAmMq.fCHXj.VuxHvx']::LTREE[])::LTREE[]::LTREE[])::LTREE[]::LTREE[] ?<@ 'foo_bar.baz'::LTREE)::LTREE
        								),
        								(
        									subpath('foo'::LTREE, array_position(ARRAY['givJaRIa.IZPIcBCTo','QbRJzQCNN.nhkBypM','Djx.qwepwR.KXkeEcaa.xbbqfa.Tf.iTfLZTNa']::LTREE[], 'wLEKwf'::LTREE, (-5662965464532109142):::INT8::INT8)::INT8::INT8)::LTREE
        								)
        						)
        							AS "t-😿a\nb_17580" (col😸_43504)
        				),
        			with_7150 ("co'l_43505")
        				AS (
        					SELECT
        						*
        					FROM
        						(VALUES (false), (false), ((NOT NULL)), (true), ((NOT true))) AS tab_17581 ("co'l_43505")
        				)
        		SELECT
        			tab_17578.crdb_internal_mvcc_timestamp AS col_43506
        		FROM
        			with_7149 AS cte_ref_2115
        		WHERE
        			false
        		LIMIT
        			1:::INT8
        	)
        		AS col_43507,
        	(-0.3115716278553009):::FLOAT8 AS col_43508,
        	tab_17578._int2 AS "%q�c😈ol_43509",
        	NULL AS col_43510,
        	NULL AS col_43511,
        	tab_17578._interval AS col_43512
        FROM
        	defaultdb.public.seed@seed__int8__float8__date_idx AS tab_17578,
        	defaultdb.public.seed@seed__int8__float8__date_idx AS "t̄a%pb_17579"
        WHERE
        	NULL
        LIMIT
        	92:::INT8;
```
Schema:

```
    rsg_test.go:783: To reproduce, use schema:
    rsg_test.go:785: SET sql_safe_updates = false;;
    rsg_test.go:785: SET CLUSTER SETTING sql.stats.automatic_collection.enabled = false;;
    rsg_test.go:785: SET CLUSTER SETTING sql.stats.histogram_collection.enabled = false;;
    rsg_test.go:785: SET CLUSTER SETTING sql.stats.forecasts.enabled = false;;
    rsg_test.go:785: CREATE TABLE "ta/b😏le_1" (col1_0 BIT(28) NULL, co😛l1_1 BIT(0) NOT NULL, "c%vol1_2" BIT(34) NULL, col1_3 BOOL NULL, col1_4 INT4 NOT NULL, "col😺1 _5" REGTYPE, col1_6 BIT(17) NOT NULL, col1_7 DECIMAL[] NULL, col1_8 OID NULL, col1_9 VARCHAR NULL, "col1%p_10" STRING NULL AS (lower(CAST(col1_3 AS STRING))) STORED, col1_11 STRING NOT NULL AS (lower(CAST(co😛l1_1 AS STRING))) STORED, "co""l1_12" STRING NOT NULL AS (lower(CAST(col1_6 AS STRING))) VIRTUAL, col1_13 STRING NULL AS (lower(CAST(col1_3 AS STRING))) STORED, col1😎_14 STRING NOT NULL AS (lower(CAST(co😛l1_1 AS STRING))) VIRTUAL, "co\gl1_15" STRING NULL AS (lower(CAST(col1_3 AS STRING))) STORED, col1_16 STRING NOT NULL AS (lower(CAST(col1_8 AS STRING))) VIRTUAL, PRIMARY KEY (col1_16 DESC), UNIQUE (co😛l1_1 DESC) STORING ("c%vol1_2", col1_9, col1_11, col1_13), UNIQUE (col1_0) WHERE ((((((((("ta/b😏le_1"."col1%p_10" < '"':::STRING) OR ("ta/b😏le_1".col1_4 = 0:::INT8)) AND ("ta/b😏le_1".col1_11 = '':::STRING)) OR ("ta/b😏le_1".col1😎_14 < e'\U00002603':::STRING)) OR ("ta/b😏le_1".col1_13 = e'\x00':::STRING)) AND ("ta/b😏le_1".col1_16 != '"':::STRING)) AND ("ta/b😏le_1".col1_9 > '"':::STRING)) OR "ta/b😏le_1".col1_3) AND ("ta/b😏le_1"."co\gl1_15" = e'\U00002603':::STRING)) AND ("ta/b😏le_1"."co""l1_12" <= e'\x00':::STRING), UNIQUE (lower(CAST(col1_3 AS STRING)) DESC) WHERE (((((((("ta/b😏le_1".col1_13 = e'\x00':::STRING) OR ("ta/b😏le_1"."co\gl1_15" = e'\x00':::STRING)) OR ("ta/b😏le_1".col1😎_14 != 'X':::STRING)) AND ("ta/b😏le_1".col1_16 < '':::STRING)) AND ("ta/b😏le_1".col1_9 = '':::STRING)) AND ("ta/b😏le_1"."col1%p_10" >= e'\x00':::STRING)) AND ("ta/b😏le_1"."co""l1_12" != '':::STRING)) AND ("ta/b😏le_1".col1_11 > '':::STRING)) OR ("ta/b😏le_1".col1_4 = 127:::INT8), UNIQUE ((CASE WHEN "col😺1 _5" IS NULL THEN e'S?s[\x10_$':::STRING ELSE e'\x0e<\x1b"\x06':::STRING END)) WHERE (((((((((NOT "ta/b😏le_1".col1_3) AND ("ta/b😏le_1"."co\gl1_15" < 'X':::STRING)) AND ("ta/b😏le_1".col1_9 != '':::STRING)) OR ("ta/b😏le_1".col1_16 >= e'\U00002603':::STRING)) AND ("ta/b😏le_1"."co""l1_12" != 'X':::STRING)) AND ("ta/b😏le_1"."col1%p_10" = e'\U00002603':::STRING)) AND ("ta/b😏le_1".col1_11 > '"':::STRING)) AND ("ta/b😏le_1".col1_4 = (-1):::INT8)) OR ("ta/b😏le_1".col1_13 > e'\'':::STRING)) AND ("ta/b😏le_1".col1😎_14 < '':::STRING), FAMILY (col1_7), FAMILY (col1_11, col1_3), FAMILY (col1_4), FAMILY (co😛l1_1), FAMILY ("col1%p_10"), FAMILY (col1_8), FAMILY (col1_13), FAMILY ("col😺1 _5"), FAMILY ("c%vol1_2"), FAMILY (col1_9, col1_6), FAMILY (col1_0, "co\gl1_15"));
    rsg_test.go:785: ALTER TABLE "ta/b😏le_1" INJECT STATISTICS e'[{"avg_size": 9, "columns": ["col1_0"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 44682978826, "histo_buckets": [{"distinct_range": 0, "num_eq": 70, "num_range": 0, "upper_bound": "0000000000000000000000000000"}, {"distinct_range": 3878252653.051425, "num_eq": 5374499426522046661, "num_range": 8000000000, "upper_bound": "0111100100111011010000101110"}, {"distinct_range": 3241874220579689, "num_eq": 5967937672457791079, "num_range": 1115222931913227518, "upper_bound": "0111110010111010101000001100"}, {"distinct_range": 7717275015713900000, "num_eq": 7485370710062056597, "num_range": 7717275015713899652, "upper_bound": "1011000010010100111101001110"}], "histo_col_type": "BIT(28)", "histo_version": 3, "name": "__auto__", "null_count": 36962539768, "row_count": 60000000000}, {"avg_size": 2, "columns": ["co\U0001F61Bl1_1"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 36770053061, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 60000000000}, {"avg_size": 22, "columns": ["\\"c%vol1_2\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 141967822, "histo_col_type": "", "name": "__auto__", "null_count": 34113706555, "row_count": 60000000000}, {"avg_size": 6, "columns": ["col1_3"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 10618632762, "histo_col_type": "", "name": "__auto__", "null_count": 8704316614, "row_count": 60000000000}, {"avg_size": 14, "columns": ["col1_4"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 26782778046, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 60000000000}, {"avg_size": 8, "columns": ["\\"col\U0001F63A1 _5\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 17217430496, "histo_col_type": "", "name": "__auto__", "null_count": 23400981747, "row_count": 60000000000}, {"avg_size": 1, "columns": ["col1_6"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 10459283209, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 60000000000}, {"avg_size": 19, "columns": ["col1_7"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 44296018197, "histo_col_type": "", "name": "__auto__", "null_count": 27521430855, "row_count": 60000000000}, {"avg_size": 9, "columns": ["col1_8"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 32228887265, "histo_col_type": "", "name": "__auto__", "null_count": 46543470224, "row_count": 60000000000}, {"avg_size": 28, "columns": ["col1_9"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 54838621732, "histo_col_type": "", "name": "__auto__", "null_count": 16479618152, "row_count": 60000000000}, {"avg_size": 30, "columns": ["\\"col1%p_10\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 53792688809, "histo_col_type": "", "name": "__auto__", "null_count": 5202559303, "row_count": 60000000000}, {"avg_size": 3, "columns": ["col1_11"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 17047451162, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 60000000000}, {"avg_size": 11, "columns": ["\\"co\\"\\"l1_12\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 14652205985, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 60000000000}, {"avg_size": 19, "columns": ["col1_13"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 28710057558, "histo_col_type": "", "name": "__auto__", "null_count": 59361687781, "row_count": 60000000000}, {"avg_size": 21, "columns": ["col1\U0001F60E_14"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 30986338795, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 60000000000}, {"avg_size": 15, "columns": ["\\"co\\\\gl1_15\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 25892506863, "histo_col_type": "", "name": "__auto__", "null_count": 55710714102, "row_count": 60000000000}, {"avg_size": 27, "columns": ["col1_16"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 38255519483, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 60000000000}]':::JSONB;
    rsg_test.go:785: CREATE TYPE rand_typ_0 AS ENUM ('aglwy', 'svxbjn', 'yeg', 'sgtd', 'zktlw');
    rsg_test.go:785: CREATE TYPE rand_typ_1 AS ENUM ('banx', 'vb', 'z', 'let');
    rsg_test.go:785: SET sql_safe_updates = false;;
    rsg_test.go:785: 
        BEGIN; CREATE TYPE greeting AS ENUM ('hello', 'howdy', 'hi', 'good day', 'morning'); COMMIT;
        BEGIN;
        CREATE TABLE IF NOT EXISTS seed AS
        	SELECT
        		g::INT2 AS _int2,
        		g::INT4 AS _int4,
        		g::INT8 AS _int8,
        		g::FLOAT4 AS _float4,
        		g::FLOAT8 AS _float8,
        		'2001-01-01'::DATE + g AS _date,
        		'2001-01-01'::TIMESTAMP + g * '1 day'::INTERVAL AS _timestamp,
        		'2001-01-01'::TIMESTAMPTZ + g * '1 day'::INTERVAL AS _timestamptz,
        		g * '1 day'::INTERVAL AS _interval,
        		g % 2 = 1 AS _bool,
        		g::DECIMAL AS _decimal,
        		g::STRING AS _string,
        		g::STRING::BYTES AS _bytes,
        		substring('00000000-0000-0000-0000-' || g::STRING || '00000000000', 1, 36)::UUID AS _uuid,
        		'0.0.0.0'::INET + g AS _inet,
        		g::STRING::JSONB AS _jsonb,
        		enum_range('hello'::greeting)[g] as _enum
        	FROM
        		generate_series(1, 5) AS g;
        COMMIT;
        
        INSERT INTO seed DEFAULT VALUES;
        CREATE INDEX on seed (_int8, _float8, _date);
        CREATE INVERTED INDEX on seed (_jsonb);
        ;
    rsg_test.go:785: ALTER TABLE seed SPLIT AT VALUES (2219519821269877619);
    rsg_test.go:785: ALTER TABLE seed SPLIT AT VALUES (617438402117064656);
    rsg_test.go:785: ALTER TABLE seed SPLIT AT VALUES (685849287110684778);
    rsg_test.go:785: ALTER TABLE seed SPLIT AT VALUES (721766062138386774);
    rsg_test.go:785: ALTER TABLE seed SPLIT AT VALUES (2333389703447591036);
    rsg_test.go:785: ALTER TABLE seed SPLIT AT VALUES (4722985919641811203);
    rsg_test.go:785: ALTER TABLE seed SPLIT AT VALUES (721487351414902396);
    rsg_test.go:785: ALTER TABLE seed SPLIT AT VALUES (363055580220680222);
    rsg_test.go:785: ALTER TABLE seed SPLIT AT VALUES (9003637191489402245);
    rsg_test.go:785: ALTER TABLE seed SCATTER;
    rsg_test.go:787: 
    rsg_test.go:788: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/d437d2c847dfedbc4972f231c3331c8e/logTestRandomSyntaxSQLSmith230527674
--- FAIL: TestRandomSyntaxSQLSmith (303.09s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #152237 sql/tests: TestRandomSyntaxSQLSmith failed [range operator should always have tight constraints] [C-test-failure O-robot T-sql-queries branch-release-25.3.0-rc release-blocker]
- #152004 sql/tests: TestRandomSyntaxSQLSmith failed [C-test-failure O-robot T-sql-queries branch-release-24.1.23-rc]
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxSQLSmith.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

sql/tests.TestRandomSyntaxSQLSmith [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/20382464?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/20382464?buildTab=artifacts#/) on master @ [f8a09a5c1dbcbfcb537ba4050c6061ef6d917687](https://github.com/cockroachdb/cockroach/commits/f8a09a5c1dbcbfcb537ba4050c6061ef6d917687):

Random syntax error:

```
    rsg_test.go:982: Crash detected: server panic: pq: internal error: expected *DLTree, found tree.dNull
```
Query:

```
        WITH
        	with_60112 (co͋l_364422) AS (SELECT * FROM (VALUES ((true AND false))) AS tab_147753 (co͋l_364422)),
        	wit😮h_60113 ("co'l%p_364423")
        		AS (
        			SELECT
        				*
        			FROM
        				(
        					VALUES
        						('HmynrO.Jlca.OEzgbmq.dHJzgk.dMibYfj.BNHhILax.BFwtjMnzV.zR'),
        						(
        							(((ARRAY['TrZDfhM.tOdGEpS.burp.lmTsjD','']::LTREE[] ?<@ 'b.Kfldu.OMWlLX.mLTh.HQJnWdfA.fptwzmcML.hjRnrxfGe'::LTREE)::LTREE::LTREE || ((NULL::LTREE || ARRAY['k.Fr.xRPbMqfVh.ajB.c.gpS.ZnTfssGC.FrAk.qn']::LTREE[])::LTREE[]::LTREE[] ?@> 'OOIW.zBmN'::LTREE)::LTREE::LTREE)::LTREE::LTREE || 'dCpYoGb.KFhZt.JLLtU'::LTREE)::LTREE
        						)
        				)
        					AS tab_147754 ("co'l%p_364423")
        		)
        SELECT
        	tab_147755.col1_1 AS col_364424
        FROM
        	defaultdb.public.table_1 AS tab_147755
        ORDER BY
        	tab_147755.col1_5 ASC NULLS LAST,
        	tab_147755.col1_5 ASC NULLS FIRST,
        	tab_147755."col1'_13",
        	tab_147755.col1_2 NULLS LAST,
        	tab_147755."col1 _12" DESC
        LIMIT
        	94:::INT8;
```
Schema:

```
    rsg_test.go:783: To reproduce, use schema:
    rsg_test.go:785: SET sql_safe_updates = false;;
    rsg_test.go:785: SET CLUSTER SETTING sql.stats.automatic_collection.enabled = false;;
    rsg_test.go:785: SET CLUSTER SETTING sql.stats.histogram_collection.enabled = false;;
    rsg_test.go:785: SET CLUSTER SETTING sql.stats.forecasts.enabled = false;;
    rsg_test.go:785: CREATE TABLE table_1 (col1_0 REFCURSOR NOT NULL, col1_1 "char" NOT NULL, col1_2 REGROLE, col1_3 TIMESTAMP NOT NULL, col1_4 INET NOT NULL, col1_5 LTREE NOT NULL, col1_6 BIT(4) NOT NULL, col1_7 NAME, "c%p
ol1'_8" GEOGRAPHY, col1_9 STRING NOT NULL, col1_10 BIT(42) NOT NULL, "col\\u38531_11" TIMESTAMPTZ, "col1 _12" STRING NOT NULL AS (lower(CAST(col1_5 AS STRING))) VIRTUAL, "col1'_13" STRING NOT NULL AS (CASE WHEN col1_3 IS NULL THEN 'U':::STRING ELSE e':n[^\b}x':::STRING END) VIRTUAL, "�col1_14" STRING NOT NULL AS (lower(CAST(col1_10 AS STRING))) STORED, PRIMARY KEY ("�col1_14" DESC, col1_5, col1_4), INDEX (col1_7 ASC, col1_4 ASC) STORING (col1_0, col1_1, col1_2, col1_3, "c%p
ol1'_8", "col\\u38531_11") NOT VISIBLE, UNIQUE (col1_4 ASC) PARTITION BY LIST (col1_4) (PARTITION table_1_part_0 VALUES IN (('cdc0:dc66:7fdd:1bf:3353:e8f1:f79e:8b3f/59':::INET,), ('139.211.65.176/15':::INET,), ('986d:46f9:7c6f:30d7:76ae:bda8:2b5:7fbf':::INET,), ('14.234.12.209/6':::INET,), ('c045:61b4:ab6c:5478:949c:9ec7:26c4:3ac0/21':::INET,), ('122.6.187.29/26':::INET,), ('ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff':::INET,), ('313e:86f0:2ce5:fc3b:67c3:5c8:c6c8:a3b5/34':::INET,), ('149.178.208.71/21':::INET,), ('41cc:2474:d106:8a0b:e5e1:749f:1726:2cf5/36':::INET,)), PARTITION table_1_part_1 VALUES IN (('881e:5227:bf56:e97e:aef2:f85:32ab:4a2/0':::INET,), ('3841:96e0:b914:a190:a786:db22:55a1:d2/86':::INET,), ('9e:50d8:6803:6da8:f935:ee8:fb7d:bea9/37':::INET,), ('89.75.221.211/29':::INET,), ('faa9:53ca:c40:5fa5:29b3:a0d8:c9d1:7c46/94':::INET,), ('f76e:fb93:36f5:5c4a:b787:7af1:376f:9535/8':::INET,), ('0.0.0.0/0':::INET,), ('211.180.197.197/20':::INET,), ('9ffd:35d3:591f:f182:8eea:a274:b2a4:a8f5/33':::INET,), ('5370:f273:3dd1:92eb:478f:3018:4d5e:a6f2/96':::INET,)), PARTITION table_1_par😠t_2 VALUES IN (('ad27:5f89:2afa:6bd6:4284:637f:6df:ea68/45':::INET,), ('9.122.99.10/14':::INET,), ('e67b:b3be:1320:2db9:ea19:3d34:309a:9436/127':::INET,), ('135.4.222.101/29':::INET,), ('100.231.227.142/14':::INET,), ('f664:983a:7b14:7661:7710:ed4f:7173:7d6/9':::INET,), ('248.179.160.135/6':::INET,))) WHERE (((table_1.col1_1 < e'\x00':::STRING) AND (table_1."�col1_14" < 'X':::STRING)) AND (table_1.col1_7 <= e'\'':::STRING:::NAME)) OR (table_1.col1_3 >= '294276-12-31 23:59:59.999999':::TIMESTAMP), INVERTED INDEX (col1_2 ASC, col1_7 gin_trgm_ops) WHERE (((((((table_1."�col1_14" = '"':::STRING) OR (table_1.col1_3 <= '3000-01-01 00:00:00':::TIMESTAMP)) OR (table_1.col1_7 <= e'\'':::STRING:::NAME)) AND (table_1."col1 _12" != e'\U00002603':::STRING)) OR (table_1.col1_9 = 'X':::STRING)) AND (table_1."col1'_13" <= 'X':::STRING)) OR (table_1."col\\u38531_11" > '3000-01-01 00:00:00+00':::TIMESTAMPTZ)) AND (table_1.col1_1 >= e'\x00':::STRING), FAMILY (col1_2, col1_0, col1_7, col1_10), FAMILY (col1_6, col1_1, col1_9), FAMILY ("�col1_14", col1_5), FAMILY ("c%p
ol1'_8", col1_4), FAMILY ("col\\u38531_11", col1_3));
    rsg_test.go:785: ALTER TABLE table_1 INJECT STATISTICS e'[{"avg_size": 31, "columns": ["col1_0"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 85414104199724927, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 1577129434965723009}, {"avg_size": 5, "columns": ["col1_1"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 188477260510000182, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 1577129434965723009}, {"avg_size": 16, "columns": ["col1_2"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1424104584800097008, "histo_buckets": [{"distinct_range": 0, "num_eq": 1000000000, "num_range": 0, "upper_bound": "0"}, {"distinct_range": 6965363036947701000, "num_eq": 3423461133745231998, "num_range": 6965363036947700703, "upper_bound": "3179511700"}, {"distinct_range": 9202741015934952000, "num_eq": 3000, "num_range": 9202741015934952879, "upper_bound": "1157286364"}, {"distinct_range": 90, "num_eq": 10000, "num_range": 90, "upper_bound": "538214174"}, {"distinct_range": 2354398776112949000, "num_eq": 5684878968954052394, "num_range": 2354398776112949201, "upper_bound": "1512480479"}, {"distinct_range": 1710541.9786706972, "num_eq": 5432079452860113599, "num_range": 2000000, "upper_bound": "3331301859"}, {"distinct_range": 0, "num_eq": 5336368730033182141, "num_range": 0, "upper_bound": "2523326568"}, {"distinct_range": 0, "num_eq": 8797708644111201487, "num_range": 3545675247181661628, "upper_bound": "2969649079"}], "histo_col_type": "REGROLE", "histo_version": 3, "name": "__auto__", "null_count": 744902152884034082, "row_count": 1577129434965723009}, {"avg_size": 10, "columns": ["col1_3"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1399658652006151963, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 1577129434965723009}, {"avg_size": 11, "columns": ["col1_4"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 944686089801693678, "histo_buckets": [{"distinct_range": 0, "num_eq": 700, "num_range": 0, "upper_bound": "43.155.241.8/3"}], "histo_col_type": "INET", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 1577129434965723009}, {"avg_size": 11, "columns": ["col1_5"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 993519053345442295, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 1577129434965723009}, {"avg_size": 21, "columns": ["col1_6"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 884964122089264943, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 1577129434965723009}, {"avg_size": 26, "columns": ["col1_7"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1066000308979003199, "histo_buckets": [{"distinct_range": 0, "num_eq": 3696180480967258785, "num_range": 0, "upper_bound": "\\u0003"}, {"distinct_range": 0, "num_eq": 40000000000, "num_range": 1000000, "upper_bound": "\\t*l"}, {"distinct_range": 100000, "num_eq": 400, "num_range": 100000, "upper_bound": "wm%"}, {"distinct_range": 0, "num_eq": 400000000, "num_range": 3953998949240096503, "upper_bound": "\\u0014$\\")="}], "histo_col_type": "NAME", "histo_version": 3, "name": "__auto__", "null_count": 1402507115045441980, "row_count": 1577129434965723009}, {"avg_size": 31, "columns": ["\\"c%p\\rol1\'_8\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1222700862708059842, "histo_col_type": "", "name": "__auto__", "null_count": 1160911839107967670, "row_count": 1577129434965723009}, {"avg_size": 30, "columns": ["col1_9"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1265202532205720780, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 1577129434965723009}, {"avg_size": 23, "columns": ["col1_10"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 969715129492185974, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 1577129434965723009}, {"avg_size": 15, "columns": ["\\"col\\\\\\\\u38531_11\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 423783491201455292, "histo_col_type": "", "name": "__auto__", "null_count": 797280504803850976, "row_count": 1577129434965723009}, {"avg_size": 25, "columns": ["\\"col1 _12\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 885808819663244314, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 1577129434965723009}, {"avg_size": 0, "columns": ["\\"col1\'_13\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 171677658492045198, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 1577129434965723009}, {"avg_size": 11, "columns": ["\\"\\u000bcol1_14\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1253428605101961269, "histo_buckets": [{"distinct_range": 0, "num_eq": 600, "num_range": 0, "upper_bound": ""}, {"distinct_range": 43627.94020762436, "num_eq": 8519841912333289607, "num_range": 70000, "upper_bound": "\\u000e\\u0001ng|\\u0016"}, {"distinct_range": 66089138.62235941, "num_eq": 30000000000, "num_range": 90000000, "upper_bound": "\\u0006$5f8\\u0003I^W"}, {"distinct_range": 50239.06747529968, "num_eq": 7095900538557799222, "num_range": 100000, "upper_bound": "m\'~H:;t%\\u001d"}], "histo_col_type": "STRING", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 1577129434965723009}]':::JSONB;
    rsg_test.go:785: CREATE TYPE rand_typ_0 AS (zgj BIT(32));
    rsg_test.go:785: CREATE TYPE rand_typ_1 AS ENUM ('b', 'navz');
    rsg_test.go:785: CREATE TYPE rand_typ_2 AS ENUM ('bmjzb', 'zdruxo', 'gx');
    rsg_test.go:785: CREATE TYPE rand_typ_3 AS ENUM ('pr', 'i', 'r');
    rsg_test.go:785: CREATE TYPE rand_typ_4 AS (hci REFCURSOR, uqqt OID, bky STRING, n TIMESTAMP, k INTERVAL);
    rsg_test.go:785: SET sql_safe_updates = false;;
    rsg_test.go:785: 
        BEGIN; CREATE TYPE greeting AS ENUM ('hello', 'howdy', 'hi', 'good day', 'morning'); COMMIT;
        BEGIN;
        CREATE TABLE IF NOT EXISTS seed AS
        	SELECT
        		g::INT2 AS _int2,
        		g::INT4 AS _int4,
        		g::INT8 AS _int8,
        		g::FLOAT4 AS _float4,
        		g::FLOAT8 AS _float8,
        		'2001-01-01'::DATE + g AS _date,
        		'2001-01-01'::TIMESTAMP + g * '1 day'::INTERVAL AS _timestamp,
        		'2001-01-01'::TIMESTAMPTZ + g * '1 day'::INTERVAL AS _timestamptz,
        		g * '1 day'::INTERVAL AS _interval,
        		g % 2 = 1 AS _bool,
        		g::DECIMAL AS _decimal,
        		g::STRING AS _string,
        		g::STRING::BYTES AS _bytes,
        		substring('00000000-0000-0000-0000-' || g::STRING || '00000000000', 1, 36)::UUID AS _uuid,
        		'0.0.0.0'::INET + g AS _inet,
        		g::STRING::JSONB AS _jsonb,
        		enum_range('hello'::greeting)[g] as _enum
        	FROM
        		generate_series(1, 5) AS g;
        COMMIT;
        
        INSERT INTO seed DEFAULT VALUES;
        CREATE INDEX on seed (_int8, _float8, _date);
        CREATE INVERTED INDEX on seed (_jsonb);
        ;
    rsg_test.go:785: ALTER TABLE seed SPLIT AT VALUES (8311024717750424617);
    rsg_test.go:785: ALTER TABLE seed SPLIT AT VALUES (4399439160944528147);
    rsg_test.go:785: ALTER TABLE seed SPLIT AT VALUES (9099956350364537218);
    rsg_test.go:785: ALTER TABLE seed SPLIT AT VALUES (1695159679288467504);
    rsg_test.go:785: ALTER TABLE seed SPLIT AT VALUES (8838561207710488345);
    rsg_test.go:785: ALTER TABLE seed SPLIT AT VALUES (3494672492323006356);
    rsg_test.go:785: ALTER TABLE seed SPLIT AT VALUES (3031773546715298872);
    rsg_test.go:785: ALTER TABLE seed SPLIT AT VALUES (2072661468553106580);
    rsg_test.go:785: ALTER TABLE seed SPLIT AT VALUES (5383294847549525877);
    rsg_test.go:785: ALTER TABLE seed SCATTER;
    rsg_test.go:787: 
    rsg_test.go:788: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/d437d2c847dfedbc4972f231c3331c8e/logTestRandomSyntaxSQLSmith1731714352
--- FAIL: TestRandomSyntaxSQLSmith (303.12s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #152237 sql/tests: TestRandomSyntaxSQLSmith failed [range operator should always have tight constraints] [C-test-failure O-robot T-sql-queries branch-release-25.3.0-rc release-blocker]
- #152004 sql/tests: TestRandomSyntaxSQLSmith failed [C-test-failure O-robot T-sql-queries branch-release-24.1.23-rc]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxSQLSmith.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/sql/sem/eval/binary_op.go
+++ b/pkg/sql/sem/eval/binary_op.go
@@ -349,6 +349,9 @@ func (e *evaluator) EvalFirstContainedByLTreeOp(
 	array := tree.MustBeDArray(a)
 	elem := tree.MustBeDLTree(b)
 
+	if array.HasNulls {
+		return nil, pgerror.New(pgcode.NullValueNotAllowed, "array must not contain nulls")
+	}
 	for _, d := range array.Array {
 		if elem.LTree.Contains(tree.MustBeDLTree(d).LTree) {
 			return tree.MustBeDLTree(d), nil
@@ -408,6 +411,9 @@ func (e *evaluator) EvalFirstContainsLTreeOp(
 	array := tree.MustBeDArray(a)
 	elem := tree.MustBeDLTree(b)
 
+	if array.HasNulls {
+		return nil, pgerror.New(pgcode.NullValueNotAllowed, "array must not contain nulls")
+	}
 	for _, d := range array.Array {
 		if tree.MustBeDLTree(d).LTree.Contains(elem.LTree) {
 			return tree.MustBeDLTree(d), nil

--- a/pkg/sql/sem/eval/eval_test.go
+++ b/pkg/sql/sem/eval/eval_test.go
@@ -351,6 +351,8 @@ func TestEvalError(t *testing.T) {
 		{`B'1001' & B'101'`, `cannot AND bit strings of different sizes`},
 		{`B'1001' | B'101'`, `cannot OR bit strings of different sizes`},
 		{`B'1001' # B'101'`, `cannot XOR bit strings of different sizes`},
+		{`ARRAY['A.B.C', NULL]::LTREE[] ?@> 'A.B.C'`, `array must not contain nulls`},
+		{`ARRAY['A.B.C', NULL]::LTREE[] ?<@ 'A.B.C'`, `array must not contain nulls`},
 	}
 	ctx := context.Background()
 	for _, d := range testData {

